Four point oh

deathaxe opened the pull request:
This PR prepares package_control_channel for Package Control 4.0 by...

1. upgrading scheme version of channel and repositories to 4.0.0
2. opt-in compatible libraries to python 3.8
3. drop all ST2-only repositories/packages/libraries

This PR is frequently synced with this repo's master branch and https://github.com/packagecontrol/channel.

It contains all current ST3+ packages and libraries.

As many as possible packages have been migrated from external repositories to this one, to reduce attack surface of uncontrolled 3rd-party registries.

> **Warning**
>
> This PR must not be merged before packagecontrol.io has been upgraded
> see: 
> - https://github.com/wbond/packagecontrol.io/pull/155
> - https://github.com/wbond/packagecontrol.io/pull/156
> - https://github.com/wbond/packagecontrol.io/pull/157

deathaxe committed:
Update to scheme 4.0.0

1. rename dependencies.json to libraries.json
2. update scheme versions
3. drops repositories of scheme 1.x as those support ST2 only.
4. drops libraries, which support ST2 only.
5. drops packages, which support ST2 only.

deathaxe committed:
Remove sublime_text key from libraries.json releases

The `sublime_text` key is optional as of PC 4.0.0-beta.9

deathaxe committed:
Assign jsonschema

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Rename bs4 to beautifulsoup4

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

deathaxe committed:
Rename python-docx to docx

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Add list of required "libraries", to ensure "lxml" is also installed.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

deathaxe committed:
Rename python-jinja2 to Jinja2

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

deathaxe committed:
Rename python-markdown to Markdown

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

deathaxe committed:
Rename python-six to six

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

Note: Opt-in to python 3.8 as it is a pure python library.

deathaxe committed:
Rename python-toml to toml

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

deathaxe committed:
Rename python-pywin32 to pywin32

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

deathaxe committed:
Rename ruamel-yaml to ruamel.yaml

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

deathaxe committed:
Fix FileBrowser "dependencies" key

deathaxe committed:
Unittests: Drop python 2 support for scheme tests

deathaxe committed:
Unittests: Drop python 2 support

deathaxe committed:
Adjust local unittests to scheme version 4.0.0

This commit ...

1. renames tokens:
   dependency => library
   dependencies => libraries

2. adjusts scheme versions to 4.0.0

3. drops `load_order` from libraries

4. drops tests for presence of `sublime_text` and `platforms` keys from release
   tests, as those are optional as of PC4.0-beta.9

5. drops duplicate library vs. package name checks, as libraries are installed
   independently to Libs/ as of PC4.0

6. adds `python_versions` type and value checks.

7. verify new libraries are not named like renamed legacy dependencies.

deathaxe committed:
Rename serial to pyserial

Rename dependency to PyPI package name to avoid future naming conflicts.

Note: Required name mapping to ensure backward compatibility is supported
      as of PC 4.0.0-beta.9

Note: Opt-in to python 3.8 as it is a pure python library.

deathaxe committed:
Unittests: Add missing Default packages

This commit adds `Binary` and `Git Formats` to the list of default packages.

deathaxe committed:
Unittests: Add missing Default packages

This commit adds `Binary` and `Git Formats` to the list of default packages.

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Unittests: Fix typo

deathaxe committed:
Unittests: Improve URL test patterns

1. sort urls alphabetically
2. reformat patterns to use more of available space
3. disallow `#` (start of anchors) and `?` (start of parameters)
   in user and repository names
4. move (?<!\.git) to ensure not to accept something like
   https://github.com/repo/user.git/tree/master

deathaxe committed:
Unittests: Add arm64 to architecture tests

Now that arm64 is supported it is very valid to have both x32 and x64
to specify arm64 is not supported.

deathaxe committed:
Unittests: Add support for explicit $schema key

To support validation and completions via LSP-json, Package Control ships
json schemas for channels and repositories.

This commit allows them to be explicitly assigned without tests failing via
- "$schema": "sublime://packagecontrol.io/schemas/channel",
- "$schema": "sublime://packagecontrol.io/schemas/repository",

deathaxe committed:
Merge remote-tracking branch 'wbond/master' into four-point-oh

deathaxe committed:
Opt-in more libraries to python 3.8

deathaxe committed:
Add pypi packages for coverage/pywinpty

Ship libraries for python 3.8 directly from pypi.

deathaxe committed:
Drop incompatible ST2 repository

Remove repository of scheme 1.2 as no longer supported.

deathaxe committed:
Drop malformed "Cross" package

Dropped package.json is missing "details" or "base" entries for more about 8
years and thus has never been crawled. As such, drop that questionable thing.

deathaxe committed:
Merge remote-tracking branch 'wbond/master' into four-point-oh

deathaxe committed:
Drop redundant code paths

url and base keys are already handled some lines above.

deathaxe committed:
Merge branch 'pr/update-tests' into four-point-oh

deathaxe committed:
Unittests: Add support for pypi provider

deathaxe committed:
Unittests: Add support for assets

deathaxe committed:
Fix platform pattern

... didn't match osx-arm64 before.

deathaxe committed:
Merge branch 'pr/update-tests' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Update libraries

Sync with intermediate channel from https://github.com/packagecontrol/channel

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Fix line ending mess on Windows

Sublime Merge messes line endings when resolving conflicts on Windows
with the removed setting being present.

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Simplify more ST version selectors

Package Control 4 requires ST3143+.
All older builds are ignored.

So `*` is equal to `>=3143`.

Selectors with lower numbers are no longer needed.

deathaxe committed:
Merge branch 'master' into cleanup

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Reapply "Remove ST2 packages (#9100)" (#9105)

This reverts commit 501d1ea6774ae8d573c594e2d9caa5b0d26db004.

deathaxe committed:
Merge branch 'prep-four-point-oh' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Migrate Test Runner package

deathaxe committed:
Migrate jshintify package

deathaxe committed:
Migrate SurroundsWith package

deathaxe committed:
Migrate SublimeManpage package

deathaxe committed:
Migrate Octopress package

deathaxe committed:
Migrate PHP Code Coverage package

deathaxe committed:
Migrate =BB= LSL package

deathaxe committed:
Migrate IMESupport package

deathaxe committed:
Migrate Solarized Toggle package

deathaxe committed:
Migrate orgmode package

deathaxe committed:
Migrate DashDoc package

deathaxe committed:
Migrate FlooBits package

deathaxe committed:
Migrate Browser Refresh package

deathaxe committed:
Migrate Ribosome package

deathaxe committed:
Migrate distractionless package

deathaxe committed:
Migrate SublimeSimpleSync and SublimeTmpl packages

deathaxe committed:
Migrate Open Finder package

deathaxe committed:
Migrate Japanize package

deathaxe committed:
Migrate Neon Color Scheme package

deathaxe committed:
Migrate Obsidian Color Scheme package

deathaxe committed:
Migrate Erlyman package

deathaxe committed:
Migrate Console API Snippets package

deathaxe committed:
Migrate Color Scheme RSE package

deathaxe committed:
Migrate Web Inspector package

deathaxe committed:
Migrate CodeComplience package

drops ST2 releases due to invalid tags/download assets.

deathaxe committed:
Migrate Language - French package

deathaxe committed:
Migrate Pandoc package

deathaxe committed:
Migrate smart less build package

deathaxe committed:
Migrate subl protocol package

deathaxe committed:
Migrate Element Finder package

drops ST2 releases

deathaxe committed:
Migrate MSBuild package

deathaxe committed:
Migrate "timonwong" channel

Migrates:
- OmniMarkupPreviewer
- ScalaFormat
- SublimeAStyleFormatter
- SublimeAlternate

deathaxe committed:
Migrate Table Editor package

deathaxe committed:
Migrate Nettuts+ Fetch and ColorPicker packages

deathaxe committed:
Migrate requirementstxt package

deathaxe committed:
Migrate SublimeREPL package

deathaxe committed:
Migrate X3-Alpha Color Scheme package

deathaxe committed:
Migrate InputHelper package

deathaxe committed:
Migrate FichteFoll repository

deathaxe committed:
Fix package order

deathaxe committed:
Fix branch entries

deathaxe committed:
Revert "Migrate =BB= LSL package"

This reverts commit 16ad9bfd99731f1ba9eff0c5392e3f7cf04073a1.

Reason: Package Name not allowed in default channel.

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into pr/replace-external-repositories

deathaxe committed:
Merge branch 'pr/replace-external-repositories' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge libraries channel

deathaxe committed:
Fix "CheckTypos" package name

Add name field to prevent unintended package renaming
due to changed repository URL.

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Update libraries

Sync with github.com/packagecontrol/channel

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Drop Tortoise package

deathaxe committed:
Adjust sublime version specifiers

Least supported version by PC4+ is ST3143. Hence no requirement to specify
lower bound "3000".

deathaxe committed:
Sync with community libraries channel

deathaxe commented:
A relevant merge commit is 9e808d553bf51d291ea778524dc72a79cf35abc5. It merges all/many commits with package migrations into four-point-oh.

deathaxe committed:
Fix Ribosome package

Convert to an asset based release to install sublime-package file
from Github releases' assets.

braver commented:
These are not overly popular packages and it wouldn't be the first time packages have been in the registry for years that are simply impossible to install or use on current versions of Sublime Text and PC. In a way I'm not even so worried about breaking them: if someone complains there is an opportunity to get it fixed.

If I'm going to help in reviewing this I'm going to have to find a way to break this down... GitHub's PR UI won't work on a changeset this large. 

I guess the first worry has to be to get the schema tests to pass?

deathaxe committed:
Drop duplicate packages

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

kaste commented:
Usually, such large changesets require that we see and review the scripts that generated them.  One after the other, e.g. "drop all packages that have a version <3000".  If we then agree that this is useful step, @braver -- the people with commit rights -- runs the script locally and merges. 

Esp. dropping is problematic as we actually want to keep track of the used names.  (Which we already have to do in the workspace.json.)  Instead marking them tombstoned/removed: datetime is in the long run preferrable.  There is always a risk to loose the workspace.json at some point, relatively unlikely that we use the registry history (this repo) as a whole.

deathaxe commented:
The deletions itself are not the problem, but the conflicts they cause with master. I therefore don't intend to put any effort in separating them.

FWIW, this PR should be merged not before old packagecontrl.io site is dropped. Hence there's probably no reason to hurry.

kaste commented:
How and where should we backup the old tombstoned names?

deathaxe committed:
Fix sublime version specifier

Co-authored-by: Koen Lageveen <braver@users.noreply.github.com>

deathaxe commented on repository/e.json:
```diff
@@ -1202,17 +1186,20 @@
-					"branch": "master"
+					"sublime_text": "*",
+					"platforms": ["osx", "linux"],
+					"tags": true
```
They are all valid PEP440 versions, which PC4 fully supports.

PC3 also supported these style of versions, regardless SemVer having been adverticed.

deathaxe committed:
Sync with community libraries channel

deathaxe committed:
Remove obsolete/optional keys from repository

In repositories of schema v4 all keys, "packages", "libraries" and "includes"
are optional. Only restriction is at least one of them to be present.

deathaxe committed:
Sort includes alphabetically

deathaxe committed:
Remove optional "sublime_text" specifier

If not specified `"sublime_text": "*"` is used by PC4.

The key is optional in schema 4.0.0

deathaxe committed:
Trim trailing whitespace

deathaxe committed:
Migrate links to https

deathaxe committed:
Use proposed schema 4.0.0 tests

deathaxe committed:
Remove optional platform keys

The platform key is only required to declare restrictions.

deathaxe committed:
Remove obsolete sublime_text version specifiers

Package Control 4 requires at least ST3143.
Schema Version 4 is supported by Package Control 4, only.

Thus any sublime_text version specifier, defining least ST builds before 3143,
can safely be removed, as it always applies.

deathaxe committed:
Convert ranges sublime_text version specifiers

Bound version specifiers targeting builds before 3143 are converted to remove
obsolete lower bound.

deathaxe committed:
Remove incompatible releases

All package releases being restricted to builds before 3143 are removed
as being ignored by PC4 anyway.

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Cleanup new packages

deathaxe committed:
Sync with community libraries channel

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Remove optional key from libraries.json

"packages" is optional in schema v4.

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Sync with libraries channel

FichteFoll commented:
FYI, I am unsubscribing from this PR. If any new developments happen, feel free to mention me again.

deathaxe committed:
Fix schema errors

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Remove obsolete keys

deathaxe committed:
Remove duplicate package

deathaxe committed:
Migrate Viv package to internal repository

deathaxe committed:
Migrate SBNF package to internal repository

deathaxe committed:
Migrate LinkedData package to internal repository

Note:

At the time of migration, external repository maintains separate release
branches for ST3 and ST4, both associated URLs of which finally point to
sublime-package files with equal content.

The sublime_version specifier of ST3 branch is invalid; targets ST3092 - 3999

Thus internal package is created with single release branch without ST version
restrictions. PC4 requires ST3143+.

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Migrate =BB= LSL package to internal repository

deathaxe committed:
Migrate DeleteBlankLines and SREC packages

A 3rd ST2-only "Code Alignment" package is dropped.

deathaxe committed:
Migrate Quokka and Wallaby packages to internal repository

At the time of migration, external repository.json points to GitHub releases
with pre-compiled .sublime-package files.

HTTP URLs are migrated to HTTPS.

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Remove obsolete st version specifiers

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Remove obsolete st version specifiers

deathaxe committed:
Merge branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Revert renaming Ribosome package

Ensure equal set of packages compared to master branch.

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Sync with libraries channel

deathaxe committed:
Merge branch 'master' into four-point-oh

deathaxe committed:
Merge remote-tracking branch 'sublimehq/master' into four-point-oh

deathaxe committed:
Sync with libraries channel
